primitives/core/hashing/src/lib.rs

```diff
@@ -35,48 +35,51 @@ pub fn blake2_512_into(data: &[u8], dest: &mut [u8; 64]) {
 
 /// Do a Blake2 512-bit hash and return result.
 pub fn blake2_512(data: &[u8]) -> [u8; 64] {
-	let mut r = [0; 64];
-	blake2_512_into(data, &mut r);
-	r
+	let mut h = blake2::Blake2b512::new();
+	h.update(data);
+	h.finalize().into()
 }
 
+type Blake2b256 = blake2::Blake2b<U32>;
+
 /// Do a Blake2 256-bit hash and place result in `dest`.
 pub fn blake2_256_into(data: &[u8], dest: &mut [u8; 32]) {
-	type Blake2b256 = blake2::Blake2b<U32>;
 	dest.copy_from_slice(Blake2b256::digest(data).as_slice());
 }
 
 /// Do a Blake2 256-bit hash and return result.
 pub fn blake2_256(data: &[u8]) -> [u8; 32] {
-	let mut r = [0; 32];
-	blake2_256_into(data, &mut r);
-	r
+	let mut h = Blake2b256::new();
+	h.update(data);
+	h.finalize().into()
 }
 
+type Blake2b128 = blake2::Blake2b<U16>;
+
 /// Do a Blake2 128-bit hash and place result in `dest`.
 pub fn blake2_128_into(data: &[u8], dest: &mut [u8; 16]) {
-	type Blake2b128 = blake2::Blake2b<U16>;
 	dest.copy_from_slice(Blake2b128::digest(data).as_slice());
 }
 
 /// Do a Blake2 128-bit hash and return result.
 pub fn blake2_128(data: &[u8]) -> [u8; 16] {
-	let mut r = [0; 16];
-	blake2_128_into(data, &mut r);
-	r
+	let mut h = Blake2b128::new();
+	h.update(data);
+	h.finalize().into()
```

Comment:
```suggestion
	Blake2b128::digest(data).into()
```

Why just not this? The same applies to all the others

Reply:
You are right.